Compression 2021 queries

paulcalvano opened the pull request:
Queries for #2160

rviscomi commented:
Thanks Paul. Could you take a look at the linter errors?

rviscomi commented on sql/2021/compression/compression_format_trend.sql:
```diff
@@ -0,0 +1,20 @@
+  COUNT(0) AS num_requests
+FROM
+  (
+    SELECT date, client, resp_content_encoding FROM `httparchive.almanac.requests_2020` WHERE DATE IN ('2020-08-01', '2019-07-01')
```
Oh I just learned that `almanac.requests` now only includes 2021 data. I had intended to grow the same `almanac` tables with each year, but I think I understand why you did it this way. Does it have to do with the new fields and incompatible schema? If so I think it should work to add the 2019/2020 data back into `requests` with NULL values for those new fields.

The way it's organized now means that all of the previous years' queries that depend on this table will be broken, so it'd be good to fix it. I think the `summary_response_bodies` table has the same issue. Could you combine the respective tables so that 2019-2021 are all in one and update this query?

paulcalvano replied:
We added a few new columns to the requests dataset for this years almanac, so I added them to a new table.  It probably makes sense to rerun the query for generating the requests table with the 2019 and 2020 datasets so we have them all in one place.

paulcalvano replied:
I updated the requests table to include the 2019-2020 data and updated this query to remove the JOIN in the inner query.

rviscomi commented on sql/2021/compression/compression_format_trend.sql:
```diff
@@ -0,0 +1,20 @@
+SELECT
+  EXTRACT(YEAR FROM date) AS year,
+  client,
+  IF(resp_content_encoding = "gzip", "Gzip", IF(resp_content_encoding = "br", "Brotli", IF(resp_content_encoding = "", "no text compression", "other"))) AS compression_type,
```
Nit: this if/else chain lends itself well to CASE/WHEN/ELSE statements instead and is a bit easier to read
```suggestion
  CASE
    WHEN resp_content_encoding = 'gzip' THEN 'Gzip'
    WHEN resp_content_encoding = 'br' THEN 'Brotli'
    WHEN resp_content_encoding = '' THEN 'no text compression'
    ELSE 'other'
  END AS compression_type,
```

rviscomi commented on sql/2021/compression/compression_formats.sql:
```diff
@@ -0,0 +1,16 @@
+  client,
+  resp_content_encoding,
+  COUNT(0) AS num_requests,
+  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client) AS pct
```
Nit: also include the total for completeness

```suggestion
  SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client) AS pct
```

rviscomi commented on sql/2021/compression/compression_formats_text_responses.sql:
```diff
@@ -0,0 +1,24 @@
+# compression_formats_text_responses.sql : What compression formats are being used (gzip, brotli, etc) on text responses
+SELECT
+  client,
+  IF(resp_content_encoding = "gzip", "Gzip", IF(resp_content_encoding = "br", "Brotli", IF(resp_content_encoding = "", "no text compression", "other"))) AS compression_type,
```
Ditto on CASE statement

rviscomi commented on sql/2021/compression/compression_formats_text_responses.sql:
```diff
@@ -0,0 +1,24 @@
+  client,
+  IF(resp_content_encoding = "gzip", "Gzip", IF(resp_content_encoding = "br", "Brotli", IF(resp_content_encoding = "", "no text compression", "other"))) AS compression_type,
+  COUNT(0) AS num_requests,
+  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client) AS pct
```
Ditto on `total`

rviscomi commented on sql/2021/compression/lighthouse_compression_byte_savings.sql:
```diff
@@ -0,0 +1,17 @@
+  _TABLE_SUFFIX AS client,
+  ROUND(CAST(JSON_EXTRACT_SCALAR(report, "$.audits.uses-text-compression.details.overallSavingsBytes") AS INT64) / 1024 / 1024) AS mbyte_savings,
+  COUNT(0) AS num_pages,
+  ROUND(COUNT(0) * 100 / SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX), 2) AS pct_pages
```
Remove percent formatting, do in Sheets instead
```suggestion
  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS pct_pages
```

rviscomi replied:
Ditto on `total`

rviscomi commented on sql/2021/compression/lighthouse_compression_byte_savings.sql:
```diff
@@ -0,0 +1,17 @@
+  mbyte_savings
+ORDER BY
+  client,
+  mbyte_savings ASC
```
`ASC` is implied
```suggestion
  mbyte_savings
```

rviscomi commented on sql/2021/compression/lighthouse_compression_scores.sql:
```diff
@@ -0,0 +1,17 @@
+  text_compression_score
+ORDER BY
+  client,
+  text_compression_score ASC
```
Implied
```suggestion
  text_compression_score
```

paulcalvano commented:
Sorry for the delay.  I updated the almanac.requests table to include the 2019 and 2020 data, and fixed the compression_trends query so that it runs against almanac.requests.
